spec/specification/json_spec.rb

```diff
@@ -305,6 +305,61 @@ module Pod
         result = Specification.from_hash(@spec.to_hash)
         result.should == @spec
       end
+
+      describe 'Swift Version Support' do
+        it 'writes swift version in singular form' do
+          @spec.swift_version = '1.0'
+          hash = @spec.to_hash
+          hash['swift_versions'].should == '1.0'
+        end
+
+        it 'writes swift version pluralized' do
+          @spec.swift_versions = ['1.0']
+          hash = @spec.to_hash
+          hash['swift_versions'].should == ['1.0']
+        end
+
+        it 'reads swift version from a string' do
+          hash = {
+            'name' => 'BananaLib',
+            'version' => '1.0',
+            'swift_versions' => '3.2',
+          }
+          result = Specification.from_hash(hash)
+          result.swift_versions.map(&:to_s).should == ['3.2']
+        end
+
+        it 'reads swift version from an array' do
+          hash = {
+            'name' => 'BananaLib',
+            'version' => '1.0',
+            'swift_versions' => %w(3.2 4.0),
+          }
+          result = Specification.from_hash(hash)
+          result.swift_versions.map(&:to_s).should == %w(3.2 4.0)
+        end
+
+        it 'is backwards compatible with pre 1.7.0 swift version' do
+          hash = {
+            'name' => 'BananaLib',
+            'version' => '1.0',
+            'swift_version' => '3.2',
+          }
+          result = Specification.from_hash(hash)
+          result.swift_versions.map(&:to_s).should == %w(3.2)
+        end
+
+        it 'combines old and new swift version declarations' do
+          hash = {
+            'name' => 'BananaLib',
+            'version' => '1.0',
+            'swift_version' => '3.2',
+            'swift_versions' => %w(4.0 4.1),
+          }
+          result = Specification.from_hash(hash)
+          result.swift_versions.map(&:to_s).should == %w(3.2 4.0 4.1)
```

Comment:
this is pretty slick! it feels like only one of these should be specified though, right? will there be JSON specs that have both?

Reply:
you are very right. I was being overly defensive and wanted a test for me to ensure I am reading both values.

Reply:
if its OK I am keeping it.